### PR TITLE
Feature/728 language clean up

### PR DESCRIPTION
**Pull request by kcinay055679**

### CODE REVIEW

File: config/locales/fr.yml [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+      to: Jusqu'à
+      today: Aujourd'hui
+      with_enddate: Avec date de fin
+      year: Année
+    devise:
+      admin: "(Admin)"
+      admin_panel: Panneau d'administration
+      devise: Devise
+      mockdata: Mockdata
+      no_omniauth_providers: An error, no Omniauth providers are available, please contact the administrator.
```
Comment (Robin481) [inline config/locales/fr.yml:106]: French much

File: lib/custom_translation_matchers/dry_crud_ti_scanner.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    end
+
+    private
+
+    def human_attribute_name_to_key_occurences(send_node:, location:) # rubocop:disable Metrics/MethodLength
+      children = Array(send_node&.children)
+      children[0]
+      children[1]
+
+      # !send_node.loc.expression.to_json.start_with?('app/helpers')
```
Comment (Robin481) [inline lib/custom_translation_matchers/dry_crud_ti_scanner.rb:23]: ?

File: spec/features/cv_search_spec.rb [hunk 3]
```diff
@@ -60,8 +66,9 @@
       expect(page).to have_content("Keine Resultate")
       page.check('search_skills')
       expect(page).not_to have_content("Keine Resultate")
-      check_search_results(I18n.t("cv_search.skills"))
-      expect(page).to have_link(href: "#{person_people_skills_path(person)}?q=#{skill_title.split(" ").join("+")}&rating=1")
+      check_search_results(Skill.model_name.human(count: 2))
```
Comment (Robin481) [inline spec/features/cv_search_spec.rb:69]: again the count thing

File: spec/features/people_skill_search_spec.rb [hunk 4]
```diff
@@ -33,9 +33,6 @@
       visit(people_skills_path)
       fill_out_row("JUnit", 5, 3)
       add_and_fill_out_row("Rails", 4, 5)
-      add_and_fill_out_row("ember", 5, 4)
```
Comment (Robin481) [inline spec/features/people_skill_search_spec.rb:36]: why remove this without seemingly any connection?

File: spec/features/people_spec.rb [hunk 5]
```diff
@@ -20,14 +20,14 @@
       visit people_path
       bob = people(:bob)
       select_from_slim_select("#person_id_person", bob.name)
-      expect(page).to have_current_path(person_path(bob))
+      expect(page).to have_current_path(person_path(bob, locale: nil))
```
Comment (Robin481) [inline spec/features/people_spec.rb:23]: Why do we now need this?

File: spec/features/people_spec.rb [hunk 6]
```diff
@@ -314,7 +314,8 @@ def add_language(language)
 
     it 'should display message when no skills are rated' do
       visit person_path(longmax)
-      expect(page).to have_selector('p.alert.alert-info.d-flex.justify-content-between', text: I18n.t('profile.no_skills_rated_msg'))
+      error_msg = "Dieses Profil hat noch keine bewerteten Skills. Gehe zum Skills Tab und bewerte die für dich relevanten Skills."
```
Comment (Robin481) [inline spec/features/people_spec.rb:317]: Why can't we use the translation now?

**Review by Robin481 — CHANGES_REQUESTED** [review]
There's a lot to improve...

Furthermore 
1) the CV Search is also broken and always redirects to a Person's Skills instead of the place where the result was found... And the translation is also broken there.
2) The export Icon is not fitting
3) Interest in the skill search is not translated
4) Admin Panel is not translated
5) The CV / Skills tab don't work when opening a profile

File: app/views/people_skills/_person_skill.html.haml [hunk 7]
```diff
@@ -1,15 +1,15 @@
...
-  %label.form-label.text-gray#skill-level-label= t "people-skills.levels.#{ExpertiseTopicSkillValue.skill_levels.key(person_skill.level - 1)}"
+  %label.form-label.text-gray#skill-level-label= ti "people_skills.levels.#{ExpertiseTopicSkillValue.skill_levels.key(person_skill.level - 1)}"
   %input#people_skill_level{value: person_skill.level, class: "form-range w-75", type: "range", min: 1, max: 5, disabled: true}
 %div.d-flex.flex-column.align-items-center.col-3
   = render partial: "show_interest_stars", locals: { person_skill: person_skill }
 %div.d-flex.align-items-center.col-2
   %div.d-flex.flex-row.form-check
     %input.check_box#certificate-checkbox{type: "checkbox", class: "form-check-input me-2 ", checked: person_skill.certificate, disabled: true}
-    %label.form-label.text-gray= t "people-skills.certificate"
+    %label.form-label.text-gray= PeopleSkill.human_attribute_name(:certificate)
```
[the remark this answers is not in the document]
Comment (kcinay055679) [reply app/views/people_skills/_person_skill.html.haml:11]: wpu ds haut ActiveRecord isch.
ActiveRecord isch kuhl

File: config/locales/crud.de.yml [hunk 8]
```diff
@@ -1,64 +1,48 @@
...
+      none_available: "(keine verfügbar)"
       please_select: Bitte auswählen
-
     button:
-      save: Speichern
       cancel: Abbrechen
+      save: Speichern
       search: Suchen
-
+    confirm_delete: Wollen Sie diesen Eintrag wirklich löschen?
```
[the remark this answers is not in the document]
Comment (kcinay055679) [reply config/locales/crud.de.yml:29]: `Ja ich will`

File: Gemfile [hunk 9]
```diff
@@ -83,3 +84,5 @@ end
 gem 'tzinfo-data', platforms: %i[mingw mswin x64_mingw jruby]
 
 gem 'rails-controller-testing', '~> 1.0'
+
+gem 'deepl-rb', '~> 2.5'
```
[the remark this answers is not in the document]
Comment (kcinay055679) [reply Gemfile:88]: No, you're right, it's not needed

File: app/helpers/date_helper.rb [hunk 10]
```diff
@@ -7,12 +7,12 @@ def date_range_label(model_with_dates)
...
 
   def months
-    t('date.month_names')
+    (1..12).map { |m| I18n.l(DateTime.parse(Date::MONTHNAMES[m]), format: '%B') }
   end
 
   def date_range_end_label(model_with_dates)
     if model_with_dates&.till_today?
-      return "- #{t('date_range_picker.today')}"
+      return "- #{ti('date.today')}"
```
[the remark this answers is not in the document]
Comment (kcinay055679) [reply app/helpers/date_helper.rb:15]: we are still using our own translation for date.today.
You can find them under gloabl.date.today in any locale file

File: app/helpers/date_helper.rb [hunk 11]
```diff
@@ -7,12 +7,12 @@ def date_range_label(model_with_dates)
   end
 
   def months
-    t('date.month_names')
+    (1..12).map { |m| I18n.l(DateTime.parse(Date::MONTHNAMES[m]), format: '%B') }
```
[the remark this answers is not in the document]
Comment (kcinay055679) [reply app/helpers/date_helper.rb:10]: changed it back :)

File: app/helpers/auth_helper.rb [hunk 12]
```diff
@@ -21,4 +21,25 @@ def find_person_by_auth_user
...
+      [e.second.titleize, e.first.downcase.to_sym]
+    end
+    languages = languages.select { |e| I18n.available_locales.include? e.second }
+    languages = languages.map do |e|
+      [e.first, url_for(locale: e.second)]
+    end
+    options_for_select(languages, url_for(locale: I18n.locale))
+  end
+
+  def first_path
```
[the remark this answers is not in the document]
Comment (kcinay055679) [reply app/helpers/auth_helper.rb:36]: I now use a REGEX

File: config/locales/de.yml [hunk 13]
```diff
@@ -1,122 +1,233 @@
...
+    widowed: verwitwet
+  people:
+    cv:
+      no_skills_rated_msg: Dieses Profil hat noch keine bewerteten Skills. Gehe zum Skills Tab und bewerte die für dich relevanten Skills.
+    export_cv:
+      export_form:
+        anonymised_cv: Anonymisierter CV
+        competences_skills: Kernkompetenzen - Skills
+        download: Herunterladen
+        location: 'Niederlassung (für Footer):'
```
[the remark this answers is not in the document]
Comment (kcinay055679) [reply config/locales/de.yml:156]: We do need them because of the :

**Comment by kcinay055679** [comment]
- [ ] the CV Search is also broken and always redirects to a Person's Skills instead of the place where the result was found... And the translation is also broken there.
- [ x] The export Icon is not fitting
- [ ] Interest in the skill search is not translated
- [ ] Admin Panel is not translated
- [ ] The CV / Skills tab don't work when opening a profile

File: app/views/people/_search.html.haml [hunk 14]
```diff
@@ -1,9 +1,8 @@
 %div.d-flex.align-items-center.justify-content-between
   %div.d-flex.col-9.gap-3
     %span.col-6{"data-controller": "dropdown"}
-      = collection_select :person_id, :person, Person.all.sort_by {|p| p.name.downcase }, :id, :name, select_when_availabale(person), {data:{"dropdown-target": "dropdown" , action: "change->dropdown#handleChange", value: "/people/"}}
+      = collection_select :person_id, :person, Person.all.sort_by {|p| p.name.downcase }, :path, :name, select_when_available(person&.path), {data:{"dropdown-target": "dropdown" , action: "change->dropdown#handleChange"}}
```
Comment (kcinay055679) [inline app/views/people/_search.html.haml:4]: Use options_for_select

File: spec/rails_helper.rb [hunk 15]
```diff
@@ -78,7 +78,13 @@
...
   config.include(SlimselectHelpers, type: :feature)
   config.include(PeopleSkillsHelpers, type: :feature)
+  config.include(PeopleSkillsHelpers, type: :feature)
+  config.include(UtilitiesHelpers)
 
   config.infer_spec_type_from_file_location!
   config.filter_rails_from_backtrace!
+
+  config.before(:each, type: :feature) do
+    default_url_options[:locale] = I18n.default_locale
```
Comment (kcinay055679) [inline spec/rails_helper.rb:88]: Use just I18n.locale

File: app/assets/stylesheets/crud.scss [hunk 16]
```diff
@@ -56,3 +57,5 @@ footer {
 .icon-pencil { background-image: url('assets/pencil-square.svg'); }
 .icon-list { background-image: url('assets/list.png'); }
 .icon-zoom-in { background-image: url('assets/search.svg'); }
+.icon-export { background-image: url('assets/export.svg'); }
+.icon-close { background-image: url('assets/x.svg'); }
```
Comment (Robin481) [inline app/assets/stylesheets/crud.scss:61]: Can we rename this file to `cancel.svg`  ? :grin: this way it atleast speaks to what it does

File: app/controllers/people_skills/filter_params.rb [hunk 17]
```diff
@@ -40,7 +40,7 @@ def interest_of_row(row_id)
     end
 
     def skill_of_row(row_id)
-      skill_ids.nil? ? nil : skill_ids[row_id].to_i
+      skill_ids[row_id]&.to_i
```
Comment (Robin481) [inline app/controllers/people_skills/filter_params.rb:43]: Hmmm I am not sure I understand the functionality 100% but are you sure this change is fine as it changes the behaviour a bit :thinking:

**Review by Robin481 — CHANGES_REQUESTED** [review]
Looking pretty good.

One more thing:
The languages in the dropdown should always be in their respective language. E.g. English should always be English, never Inglese, even if you are currently using Italian.

**Review by Robin481 — APPROVED** [review]

**Pull request merged**